### ISSUE TITLE
user-timing/structured-serialize-detail.any.js missing toJSON() tests

### ISSUE THREAD

**Issue by k-j-kim** [web-platform-tests/wpt#26013]
In all wpt-tested browsers, `detail` is missing from `toJSON()` output of `mark` and `measure`:
```js
window.performance.mark('A', { detail: { random: 1 } } ).toJSON();
// output: {name: "A", entryType: "mark", startTime: 104306.76999990828, duration: 0}
```
I'm not sure if this should be part of:
- https://wpt.fyi/results/user-timing/structured-serialize-detail.any.html?label=experimental&label=master&aligned
OR
- https://wpt.fyi/results/user-timing/user-timing-tojson.html?label=experimental&label=master&aligned

**Comment by sideshowbarker** [comment]
cc @maxlgu @tommckee1 @plh @caribouW3

**Comment by maxlgu** [comment]
@tdresser @npm1

**Comment by npm1** [comment]
Filed crbug.com/1138437 (AFAIK only chromium-based browsers support the `detail` field right now) and will update the toJSON test when this is fixed.

**Comment by npm1** [comment]
Looks like we actually need to also add toJSON explicitly in the spec . Right now, technically it would be calling the toJSON from PerformanceEntry, so it needs its own toJSON to be able to include detail.

**Comment by k-j-kim** [comment]
@npm1 do you mean in [User Timing Level 3](https://www.w3.org/TR/user-timing-3/)? 

It's called out in [Performance Timeline Level 2#3.2 The PerformanceEntry interface](https://www.w3.org/TR/performance-timeline-2/#the-performanceentry-interface), so I thought it was implied to be present in `PerformanceMark`/`PerformanceMeasure` interfaces which extends off `PerformanceEntry`.

**Comment by k-j-kim** [comment]
FYI, a naive polyfill for `detail` if anyone is interested in using this until its fixed: https://gist.github.com/k-j-kim/29e3da0aac1c414d6b4571b47a19d458

**Comment by npm1** [comment]
@k-j-kim what I mean is just adding the toJSON method explicitly in UserTiming since now it is meant to override the one defined in PerformanceEntry (so that it can have the detail field).

**Comment by npm1** [comment]
Hmm actually I don't know if we should include `detail` because its type is `any` and hence it does not seem to be a [JSON type](https://heycam.github.io/webidl/#idl-tojson-operation). We could use a non-default toJSON operation but I'm not sure if we would be able to produce something that makes sense, since a developer can put pretty much anything in `detail`. So I think the right decision here is to not add `detail`. @domenic what do you think?

**Comment by k-j-kim** [comment]
Should it not still attempt de-serialization and return empty if not serializable? It seems unexpected that `toJSON` will omit `detail` altogether.

**Comment by domenic** [comment]
> Hmm actually I don't know if we should include detail because its type is any and hence it does not seem to be a JSON type. We could use a non-default toJSON operation but I'm not sure if we would be able to produce something that makes sense, since a developer can put pretty much anything in detail. So I think the right decision here is to not add detail. @domenic what do you think?

Yeah, I agree the right decision is not to include detail. Since it's an unstructured type, it could be anything; JSON serialization is only for the known properties of the object that have a known shape and type.
